File: 001-math/002-Divisor game/CPP/Solution.cpp
int gcd(int A, int B) {
    return B==0? A:gcd(B,A%B);
    
}

int lcm(int A, int B){
        return (A/gcd(A,B))*B;
    
}

int Solution::solve(int A, int B, int C) {
    return A/lcm(B,C);
}

File: 001-math/002-Divisor game/Go/Solution.go
/**
 * @input A : Integer
 * @input B : Integer
 * @input C : Integer
 * 
 * @Output Integer
 */
func solve(A int , B int , C int )  (int) {
    return A/lcm(B,C)
}

func gcd(A int, B int) (int){
    if(B==0) {
        return A
    }
    return gcd(B,A%B)
}

 func lcm(A int, B int)(int){
         return (A/gcd(A,B))*B
 }
